sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/ScannerFileService.java

```diff
@@ -53,22 +55,19 @@ public boolean isSupportedAbsolute(String absolutePath) {
     Matcher matcher = DETERMINISTIC_SOURCE_PATH_PREFIX.matcher(filePath.replace('\\', '/'));
     if (matcher.find()) {
       String relativePath = matcher.replaceFirst("");
-      Iterable<InputFile> files = fileSystem.inputFiles(fileSystem.predicates().hasLanguage(languageKey));
-      int count = 0;
-      InputFile foundFile = null;
-      for (InputFile file : files) {
-        String path = file.uri().getPath();
-        if (path.endsWith(relativePath)) {
-          count++;
-          foundFile = file;
-        }
-      }
-      if (count == 1) {
+      List<InputFile> foundFiles = new ArrayList<>();
+      FilePredicates fp = fileSystem.predicates();
+
+      fileSystem.inputFiles(fp.and(fp.hasLanguage(languageKey), new RelativePathPredicate(relativePath))).forEach(foundFiles::add);
+
+      int foundFilesCount = foundFiles.size();
```

Comment:
It's used once for logic and second time for logging. I'd not use the variable to spare the line & state. The `.size()` is cheap and easy to understand.